dvc/repo/experiments/executor/base.py

```diff
@@ -248,6 +250,19 @@ def on_diverged_ref(orig_ref: str, new_rev: str):
             )
         return refs
 
+    @classmethod
+    def _validate_remotes(cls, dvc: "Repo", git_remote: Optional[str]):
+
+        if git_remote == dvc.root_dir:
```

Comment:
Is there a reason to have this explicit check? Would this qualify as a valid git remote? Is there some reason to think a user might set `DVC_EXP_GIT_REMOTE` to their local project's root dir?

Reply:
Also, why fail for an invalid Git remote but only warn for this scenario?

Reply:
`git_remote` is just a variable that is either a path/URL to a git repo (including local ones) or a remote name. If this check is hit it means it's a local path that points directly to the current DVC root directory, which in most cases is a valid git repo path.

If you do `DVC_EXP_GIT_REMOTE=.` the end result would be that we auto-push DVC cache for your local experiment runs. (The git push step becomes a no-op, the same thing that happens if you do a CLI `git push` to your own current repo directory)